src/libclipper/src/query_processor.cpp

```diff
@@ -180,36 +110,22 @@ boost::future<Response> QueryProcessor::predict(Query query) {
       }
     }
 
-    Output final_output;
-
-    if (query.selection_policy_ == "EXP3") {
-      final_output =
-          combine_predictions<Exp3Policy>(query, outputs, serialized_state);
-
-    } else if (query.selection_policy_ == "EXP4") {
-      final_output =
-          combine_predictions<Exp4Policy>(query, outputs, serialized_state);
-    } else if (query.selection_policy_ == "EpsilonGreedy") {
-      final_output = combine_predictions<EpsilonGreedyPolicy>(query, outputs,
-                                                              serialized_state);
-
-    } else if (query.selection_policy_ == "UCB") {
-      final_output =
-          combine_predictions<UCBPolicy>(query, outputs, serialized_state);
-    } else {
-      UNREACHABLE();
-    }
+    std::pair<Output, bool> final_output =
+        current_policy->combine_predictions(selection_state, query, outputs);
     std::chrono::time_point<std::chrono::high_resolution_clock> end =
```

Comment:
@Corey-Zumar Can you add a ratio counter here that tracks the fraction of predictions that use the default output?